### ISSUE TITLE
Registry.parse(): a file that INCLUDEs itself recurses without bound (RangeError; heap exhaustion on large files)

### ISSUE THREAD

**Issue by ilbui** [abaplint/abaplint#4328]
**Version:** `@abaplint/core` 2.119.14 · Node v24.15.0 (also reproduced on Node 20) · Windows 11

### Summary

When a single in-memory file is added to a `Registry` under a filename whose object name matches an `INCLUDE` statement in its own source, `Registry.parse()` resolves that `INCLUDE` to the file itself and recurses without a termination check. The process does not report a parse issue — it dies:

* on a tiny source: `RangeError: Maximum call stack size exceeded`
* on a large source: the V8 heap is exhausted and the process is killed (`FATAL ERROR: Reached heap limit Allocation failed`), because each level of the recursion allocates.

This is reachable from any tool that analyses one object at a time and names the in-memory file after the object: the source can legitimately contain an `INCLUDE` with that same name.

### Minimal reproduction (two lines of ABAP)

```js
const { Registry, MemoryFile, Config, Version } = require("@abaplint/core");

const reg = new Registry();
reg.setConfig(Config.getDefault(Version.v758));
reg.addFile(new MemoryFile("zfoo.prog.abap", "REPORT zfoo.\nINCLUDE zfoo.\n"));
reg.parse();                       // throws RangeError: Maximum call stack size exceeded
```

**Expected:** `parse()` returns, and the self-reference is reported as an issue (or simply not resolved).
**Actual:** `RangeError: Maximum call stack size exceeded`, ~76 ms, on a 27-character source.

Control (same source, different filename) — parses fine and yields 2 statements:

```js
reg.addFile(new MemoryFile("zbar.prog.abap", "REPORT zfoo.\nINCLUDE zfoo.\n"));
```

### Second symptom: large files exhaust the heap instead of throwing

The same self-reference on a larger source does not raise `RangeError` — the recursion allocates per level, so the process runs out of memory first. Reproduction with a synthetic ~400 KB source:

```js
let body = "";
for (let i = 0; body.length < 400 * 1024; i++) body += `DATA lv_filler_${i} TYPE i.\n`;
const src = "REPORT zfoo.\nINCLUDE zfoo.\n" + body;

const reg = new Registry();
reg.setConfig(Config.getDefault(Version.v758));
reg.addFile(new MemoryFile("zfoo.prog.abap", src));
reg.parse();                       // FATAL ERROR: Reached heap limit Allocation failed
```

Measured with `--max-old-space-size=2048`: the process dies after ~46 s. With the default heap it grows to several GB before dying. Either way the failure is a dead process, not a diagnosable result, which makes it hard to guard against in a service that parses arbitrary input.

### Suggested handling

Detect the self-reference while resolving `INCLUDE` — a file resolving an include to the object it is already inside — and either skip that inclusion or report it as an issue. A depth/visited-set guard in the include resolution would turn both symptoms into an ordinary finding.

### Workaround (for others hitting this)

Name the in-memory file so that no `INCLUDE` statement can name it. A prefix joined with `_`, `-` or `~` is **not** enough: for a file named like `x_zfoo.prog.abap` the engine still resolves `INCLUDE x_zfoo.` and the recursion comes back (measured on 2.119.14 with all three separators). A character that an ABAP identifier cannot contain, such as `$` (`x$zfoo.prog.abap`), does not resolve, and statement extraction is unchanged.
